src/jekyll/jk_document.rs

```diff
@@ -0,0 +1,219 @@
+                )
+            }))
+            .merge_permalink(custom_attributes.remove("permalink").and_then(|v| {
+                v.as_str().map(|s| s.to_owned())
+            }))
+            .merge_draft(custom_attributes.remove("published").and_then(|v| {
+                v.as_bool().map(|b| !b)
+            }))
+            .merge_layout(custom_attributes.remove("layout").and_then(|v| {
+                v.as_str().map(|s| s.to_owned())
+            }))
+            .merge_published_date(custom_attributes.remove("date").and_then(|d| {
+                d.as_str().and_then(datetime::DateTime::parse)
+            }))
+            .merge_custom(custom_attributes)
+    }
+}
+
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+#[derive(Serialize, Deserialize)]
+pub struct JkDocument {
+    pub front: Option<String>,
+    pub content: Option<String>,
+}
+
+impl JkDocument {
+    pub fn parse_string(doc: String) -> Result<JkDocument> {
+        let (front, content) = split_document(&doc)?;
+        if front != None {
+            Err(ErrorKind::MissingFrontStart.into())
+        } else {
+            match split_document(content)? {
+                (None, _) => Err(ErrorKind::MissingFrontmatter.into()),
+                (Some(front), content) => Ok(JkDocument {
+                    front: Some(front.to_owned()),
+                    content: Some(content.to_owned()),
+                }),
+            }
+        }
+    }
+
+    pub fn parse(source_file: &Path) -> Result<JkDocument> {
+        let doc: String = read_file(source_file)?;
+        JkDocument::parse_string(doc)
+    }
+
+    pub fn convert_front(front: Option<String>) -> Result<String> {
+        let front_value: JkFrontmatterBuilder = front
+            .map(|s| serde_yaml::from_str(&s))
+            .map_or(Ok(None), |r| r.map(Some))?
+            .unwrap_or_else(JkFrontmatterBuilder::new);
+
+        let front_builder: FrontmatterBuilder = front_value.into();
+        let front = front_builder.build()?;
+        let mut converted = serde_yaml::to_string(&front)?;
+        converted.drain(..4);
+        Ok(converted)
+    }
+
+    pub fn convert(source_file: &Path, dest_dir: &Path) -> Result<()> {
+        let doc = JkDocument::parse(source_file)?;
+        let front = JkDocument::convert_front(doc.front)?;
+        if !dest_dir.exists() {
+            create_dir_all(&dest_dir)?;
+        }
+        let dest_file = dest_dir.join(source_file.file_name().unwrap());
+        let mut dest = File::create(dest_file)?;
+        let converted = format!("{}\n---\n{}", &front, &doc.content.unwrap());
+        dest.write_all(converted.as_bytes())?;
+        Ok(())
+    }
+}
+
+pub fn convert_from_jk(source: &Path, dest: &Path) -> Result<()> {
+    if dest.is_file() {
+        Err(ErrorKind::CantOutputInFile.into())
+    } else if source.is_file() {
+        JkDocument::convert(source, dest)
+    } else if source.is_dir() {
+        for file in source.read_dir()? {
```

Comment:
Should this be recursive?

Reply:
I don't know if under `_posts` folder, jekyll can have subfolder, if yes, then yeah, it should :)

Reply:
I think it can.

Reply:
Also, there are pages not in the `_posts` folder.

You can find an example of [`WalkDir`](https://github.com/cobalt-org/cobalt.rs/blob/master/src/files.rs#L45) to handle the recursive walking for you.